relay/mock/ics.go

```diff
@@ -0,0 +1,68 @@
+package mock
```

Comment:
why do we need a separate mock for relay? 
don't we have an existing mock for IndexedChainState?

Reply:
The existing `ChainDataMock` struct has a bunch of internal logic that looks like it was written to support specific unit tests. I just need a mock implementation where I can use the pattern `when X method is called with Y arguments, return Z`.